doc/CodeReview.md

```diff
@@ -0,0 +1,36 @@
+Code Review
+===========
+
+## Code Review Practices
+
+PR to `libtock-rs` can be divided into two categories:
+
+1. **Upkeep pull requests** are minor changes to existing functionality.
+   Examples include bug fixes (that do not significantly affect APIs) and
+   documentation that describes an existing implementation.
+1. **Significant pull requests** are pull requests that are too substantial to
+   be considered upkeep pull requests. Significant pull requests may include new
+   functionality, API changes, significant refactoring, new tooling, and other
+   changes.
+
+The owners of `libtock-rs` (listed [below](#owners)) determine whether a PR is
+an upkeep PR or a significant PR. PRs should be merged by the `libtock-rs`
+owners rather than the PR's author. PRs authored by `libtock-rs` owners should
+be merged by a reviewer rather than their author.
```

Comment:
I copied the "PRs should be merged by a reviewer rather than their author" part of this policy from Tock's code review document. However, I'm concerned that reviewers are likely to forget to merge significant pull requests at the end of the 7 day waiting period. I find it much easier to remember to merge pull requests that I authored, as they are on my TODO list.

Should we instead have authors submit significant pull requests themselves? We could delegate the submission using `bors d+` and comment that they should wait until day X to merge.

Reply:
I agree that people forgetting is an issue, but in that case you can just ping the PRs. I think overall it seems better to have someone besides the author merge a PR, especially for new contributors.

Maybe you can add something saying that if reviewed and passed the time and it hasn't been merged a maintainer can merge it themselves (as a fallback).

Reply:
I feel the review process for PRs from owners and non-owners should be consistent.

I can add a paragraph recommending that authors of significant PRs add a comment when they believe the waiting period has elapsed. This would also help prevent deadlocks where the reviewers think they're waiting on the author and the author thinks the PR is ready to merge.

Do you agree?

Reply:
Sounds good to me. Most projects have things saying to ping if something isn't merged. I think it's a good strategy as it's easy for things to be lost/forgotten and telling people to ping encourages them to speak up when that happens.

Reply:
I added that paragraph.